index.html

```diff
@@ -1,390 +1,1451 @@
+
+      <pre class="js">
+        const encoder = new TextEncoder();
+        const writer = port.writable.getWriter();
+        writer.write(encoder.encode("A long message that will take..."));
+        await writer.close();
+        await port.close();
+      </pre>
+
+      To discard any unsent data the application could instead call
+      `writer.`{{WritableStreamDefaultWriter/abort()}}.
+
+      <p>
+      If a {{TransformStream}} is being piped to {{SerialPort/writable}} then
+      waiting for the {{Promise}} returned by
+      `writer.`{{WritableStreamDefaultWriter/close()}} to resolve is
+      insufficient. The application must wait for the pipe chain to close by
+      waiting for the {{Promise}} returned by {{ReadableStream/pipeTo()}} to
+      resolve instead.
+
+      <pre class="js">
+        const encoder = new TextEncoderStream();
+        const writableStreamClosed = encoder.readable.pipeTo(port.writable);
+        const writer = encoder.writable.getWriter();
+        writer.write("A long message that will take...");
+        writer.close();
+        await writableStreamClosed;
+        await port.close();
+      </pre>
+
+      If a loop is being used to read chunks from the port, as is done in
+      [[[#readable-example]]], then it must be exited before calling
+      `port.`{{SerialPort/close()}}.
+
+      <pre class="js">
+        let keepReading = true;
+        let reader;
+
+        async function readUntilClosed() {
+          while (port.readable && keepReading) {
+            reader = port.readable.getReader();
+            try {
+              while (true) {
+                const { value, done } = await reader.read();
+                if (done) {
+                  // |reader| has been canceled.
+                  break;
+                }
+                // Do something with |value|...
+              }
+            } catch (error) {
+              // Handle |error|...
+            } finally {
+              reader.releaseLock();
+            }
+          }
+
+          await port.close();
+        }
+
+        readUntilClosed();
+
+        // Sometime later...
+        keepReading = false;
+        reader.cancel();
+      </pre>
+
+      Calling `reader.`{{ReadableStreamGenericReader/cancel()}} causes the call
+      to {{ReadableStreamDefaultReader/read()}} to return immediately, exiting
+      the inner loop and calling {{ReadableStreamDefaultReader/releaseLock()}}.
+      The outer loop then exits because `keepReading` has been set to `false`
+      and with the stream unlocked `port.`{{SerialPort/close()}} can complete
+      successfully.
+    </aside>
+
+    The {{SerialPort/close()}} method steps are:
+
+    <ol>
+      <li>Let |promise| be [=a new promise=].
+      <li>
+        Let |cancelPromise:Promise| be the result of invoking
+        [=this=].{{[[readable]]}}.{{ReadableStream/cancel()}} or [=a promise
+        resolved with=] `undefined` if [=this=].{{[[readable]]}} is `null`.
```

Comment:
Don't call overridable JS methods in this way. Instead, use https://streams.spec.whatwg.org/#readablestream-cancel and https://streams.spec.whatwg.org/#writablestream-abort